### PR TITLE
Add patching in microphone requirement

### PR DESCRIPTION
**Pull request by RedBrumbler**
- Allow adding mic permission to apps that don't have it normally

### CODE REVIEW

**Comment by qe201020335** [comment]
Consider also granting the permission after installeing the patched apk? like this
https://github.com/Lauriethefish/QuestPatcher/blob/dda739b0c9f33406d79bf668e6b42e1468edee12/QuestPatcher.Core/Patching/PatchingManager.cs#L645-L658

**Comment by RedBrumbler** [comment]
I'd argue mic perms are less straightforward than file perms, so if it is up to the user to then allow mic perms at runtime id rather not automatically give that perm

File: QuestPatcher/Views/PatchingView.axaml [hunk 1]
```diff
@@ -13,17 +13,24 @@
...
+                            <TextBlock Margin="0 25 0 0">Hand Tracking Type</TextBlock>
+                            <ComboBox Margin="20 10 20 20" SelectedIndex="{Binding Config.PatchingPermissions.HandTrackingType}">
+                                <ComboBoxItem>None</ComboBoxItem>
+                                <ComboBoxItem>V1</ComboBoxItem>
+                                <ComboBoxItem>V1 High Frequency</ComboBoxItem>
+                                <ComboBoxItem>V2</ComboBoxItem>
+                            </ComboBox>
+                        </StackPanel>
+                    </StackPanel>
+                    <StackPanel Orientation="Horizontal" HorizontalAlignment="Left" VerticalAlignment="Top" Spacing="25">
```
Comment (Lauriethefish) [inline QuestPatcher/Views/PatchingView.axaml:31]: Why is the mic perms switch in its own panel?

File: QuestPatcher/Views/PatchingView.axaml [hunk 2]
```diff
@@ -13,17 +13,24 @@
...
-                <ToggleSwitch IsChecked="{Binding Config.PatchingPermissions.FlatScreenSupport}">Disable VR Requirement</ToggleSwitch>
                 <StackPanel Orientation="Vertical">
-                    <TextBlock Margin="0 25 0 0">Hand Tracking Type</TextBlock>
-                    <ComboBox Margin="20 10 20 20" SelectedIndex="{Binding Config.PatchingPermissions.HandTrackingType}">
-                        <ComboBoxItem>None</ComboBoxItem>
-                        <ComboBoxItem>V1</ComboBoxItem>
-                        <ComboBoxItem>V1 High Frequency</ComboBoxItem>
-                        <ComboBoxItem>V2</ComboBoxItem>
-                    </ComboBox>
+                    <StackPanel Orientation="Horizontal" HorizontalAlignment="Left" VerticalAlignment="Top" Spacing="25">
```
Comment (Lauriethefish) [inline QuestPatcher/Views/PatchingView.axaml:17]: What is this newly added panel for?

File: QuestPatcher.Core/Patching/PatchingManager.cs [hunk 3]
```diff
@@ -280,6 +280,12 @@ private async Task PatchManifest(ZipArchive apkArchive)
                 });
             }
 
+            if (permissions.Microphone)
+            {
+                Log.Information("Adding microphone permission request . . .");
```
Comment (Lauriethefish) [inline QuestPatcher.Core/Patching/PatchingManager.cs:285]: Should be no need for this log line imo, the permissions in the `addingPermissions` list are already logged later on.

Comment (RedBrumbler) [reply QuestPatcher.Core/Patching/PatchingManager.cs:285]: Ah, OK I guess I can remove that

File: QuestPatcher/Views/PatchingView.axaml (hunk 1 above)
Comment (RedBrumbler) [reply QuestPatcher/Views/PatchingView.axaml:31]: We ran out of horizontal space when questpatcher is at the minimum size, so I added an extra line for the extra toggle

File: QuestPatcher/Views/PatchingView.axaml (hunk 2 above)
Comment (RedBrumbler) [reply QuestPatcher/Views/PatchingView.axaml:17]: Isn't this the same thing as the other place

File: QuestPatcher/Views/PatchingView.axaml (hunk 1 above)
Comment (Lauriethefish) [reply QuestPatcher/Views/PatchingView.axaml:31]: Ah ok that makes sense

Comment (RedBrumbler) [reply QuestPatcher/Views/PatchingView.axaml:31]: I messaged you on discord about that space last week

**Pull request merged**